Research/lva-node-app/public/js/main.js

```diff
@@ -0,0 +1,667 @@
+
+/**
+* get global variables from server, Returns a promise so that functions that depend on these variables can wait
+* @returns {Promise<any>}
+*/
+function getGlobals()
+{
+  return new Promise((resolve, reject) => 
+  {
+    let request = prepareRequest(`http://localhost:${PORT}/globals`, "GET");
+    request.onreadystatechange = function () 
+    {
+      console.log("REady state = "+request.readyState+" status = "+request.status);
+      if(request.readyState == 4)
+      {
+        if(request.status == 200)
+        {
+          let response=JSON.parse(request.response);
+          console.log(response);
+          graphInstances=response.graphInstances;
+          graphTopologies=response.graphTopologies;
+          cameras=response.cameras;
+          resolve(response);
+        }
+        else
+        {
+          reject("Get Globals rejection");
+        }
+      }
+      else if(request.status != 200)
+      {
+        reject("Server not responding");
+      }
+    }
+    request.send();
+  });
+}
+
+/**
+* send current values to the server to be stored for later 
+*/
+function sendGlobals()
+{
+  let globals = 
+    {
+      graphInstances: graphInstances,
+      graphTopologies: graphTopologies,
+      cameras: cameras
+    };
+  let request = prepareRequest(`http://localhost:${PORT}/globals`, "PUT");
+  request.send(JSON.stringify(globals));
+}
+
+/**
+*  generate request with given parameters to a defined url in server. default method is POST
+* @param {string} url - @link to communicate on
+* @param {string} requestType - POST or GET request 
+* @returns {XMLHttpRequest} - generated, unsent request
+*/ 
+function prepareRequest(url, requestType="POST") 
+{
+  let request = new XMLHttpRequest();
+  request.open(requestType, url, true);
+  request.setRequestHeader('Content-Type', 'application/json; charset=UTF-8');
+  return request;
+}
+
+/**
+* delete all htmlElementents nested within the nearest parent li item
+* @param {HTMLButtonElement} htmlElement - what user clicked on
+*/
+function deleteFromParentListItem(htmlElement) 
+{
+  let id = $(htmlElement).closest('li').attr('id');
+  let item = document.getElementById(id);
+  item.parentElement.removeChild(item);
+}
+
+/**
+* create unique ID for an item
+* @param {string} nameToInclude - optional name to include in unique id
+* @returns {string} - unique ID
+*/
+function makeUniqueId(nameToInclude = "") 
+{
+  //return nameToInclude + a UUID
+  return (nameToInclude + 'xxxx-xxxxxxxxxxxx'.replace(/[x]/g, function (char) 
+  {
+    let rand = Math.random() * 16 | 0, v = char == 'x' ? rand : (rand & 0x3 | 0x8);
+    return v.toString(16);
+  }));
+}
+
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//configuration page functions
+/** 
+* grab inputs from configuration. Currently set expiration to 1 day. Calls to connect to device and IotHub to make sure your credentials are valid!
```

Comment:
where is expiration set to 1 day?

Reply:
good catch-old comment from when I was using cookies!